src/arrayMethodJoin.js

```diff
@@ -4,8 +4,34 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let str = '';
+    let separ = separator;
+    const length = this.length - 1;
+
+    if (separator === null) {
+      separ = 'null';
+    }
+
+    for (let i = 0; i < this.length; i++) {
+      switch (this[i]) {
+        case null:
+        case undefined:
+          if (i !== length) {
+            str += separ;
+          }
+          break;
+        default:
+          str += this[i];
+
+          if (i !== length) {
+            str += separ;
+          }
+          break;
```

Comment:
Will be better to use switch / case when you need to do some logic depending on some variables that has limited amount of values
In this case, if / else will be better option
Just check if element is null or undefined - add only separator
Else add element + separator (if it's not the last element)